src/NLog/LayoutRenderers/Wrappers/WhenLayoutRendererWrapper.cs

```diff
@@ -83,10 +84,40 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
             }
         }
 
+        private bool ShouldRenderInner(LogEventInfo logEvent)
+        {
+            return When == null || true.Equals(When.Evaluate(logEvent));
+        }
+
         /// <inheritdoc/>
         [Obsolete("Inherit from WrapperLayoutRendererBase and override RenderInnerAndTransform() instead. Marked obsolete in NLog 4.6")]
         protected override void TransformFormattedMesssage(StringBuilder target)
         {
         }
+
+        /// <inheritdoc />
+        public bool TryGetRawValue(LogEventInfo logEvent, out object value)
+        {
+            if (ShouldRenderInner(logEvent))
+            {
+                if (Inner == null)
+                {
+                    value = null;
+                    return false;
+                }
+
+                return Inner.TryGetRawValue(logEvent, out value);
```

Comment:
Could this also work:
```c#
value = null;
if (ShouldRenderInner(logEvent))
   return Inner?.TryGetRawValue(logEvent, out value) ?? false;
else
   return Else?.TryGetRawValue(logEvent, out value) ?? false;
```

Reply:
I think it should work, but a bit hard to read. But I see the problem of the verbose/duplicate code. Will create a helper and let's see if it's then better.

Reply:
PS: we need a helper


![image](https://user-images.githubusercontent.com/5808377/57257747-9921af80-705a-11e9-8dfb-978c0631b031.png)

Reply:
added helper. Let me know what you think